### ISSUE TITLE
[Usage] mm_projector weights go to `inf`

### ISSUE THREAD

**Issue by TheMikeMerrill** [haotian-liu/LLaVA#463]
### Describe the issue

Issue:

Hey there - huge fan of your work. 

I'm trying to integrate LLaVa into a training framework I've been running into the same issue again and again. When I try to train LLaVa outside of your framework (i.e. by loading it and passing it through my own training loop) something breaks and I get `nan` loss. 

I've managed to isolate the issue to the `mm_projector` weights, which go to `inf` after the first training step. This doesn't seem to be an issue with exploding gradients since the grad norm stays low (<10) and the issue persists even with tiny learning rates (<1e-5). 

Command:

`debug.py`:

``` python
from llava.model.builder import load_pretrained_model
import numpy as np
import torch

# Load the model
tokenizer, model, image_processor, context_len  = load_pretrained_model("liuhaotian/llava-llama-2-7b-chat-lightning-lora-preview",
                                                                            model_base = "meta-llama/Llama-2-7b-chat-hf", 
                                                                            model_name = "llava-llama-2-7b-chat-lightning-lora-preview")
# Unfreeze the mm projector
for p in model.get_model().mm_projector.parameters():
    p.requires_grad = True

optimizer = torch.optim.Adam(model.parameters(), lr=1e-3)
for i in range(10):
    
    optimizer.zero_grad()

    # Dummmy inputs
    imtensors = [np.random.rand(256,256,3)]
    input_images = image_processor(imtensors, return_tensors="pt",
                                            ).pixel_values
    input_images = torch.tensor(input_images).to(model.device).type(model.dtype)

    inputs = tokenizer(["This is a test"], return_tensors="pt").input_ids.to(model.device)
    
    results = model(input_ids = inputs, labels=inputs, images = input_images)
    print(f"Loss: {results.loss}")

    results.loss.backward()
    optimizer.step()
```
`conda activate llava`
`python debug.py`

Log: 
```
Loss: 4.4296875
Loss: nan
Loss: nan
Loss: nan
Loss: nan
Loss: nan
Loss: nan
Loss: nan
Loss: nan
Loss: nan
```

Any ideas as to why this might be happening?

**Comment by TheMikeMerrill** [comment]
@haotian-liu I'm sure you're very busy with all these issues, I'm still struggling with this and it would mean a lot if you could take a look - even just to point me in a new direction. My advisor has been asking lots of questions about it and it's been hard to make progress :(

**Comment by haotian-liu** [comment]
Thank you for your interest in LLaVA, and sorry for the delayed response.

You would probably need to set `model.to(dtype=torch.bfloat16)` or `model.to(dtype=torch.float32)`, as float16 may have gradient overflow.

**Comment by TheMikeMerrill** [comment]
That seems to have done the trick. Thank you!
